pkg/webhook/webhook.go

```diff
@@ -105,45 +103,55 @@ func (pwh *pravegaWebhookHandler) mutatePravegaVersion(ctx context.Context, p *p
 	}
 
 	requestVersion := p.Spec.Version
-	// Check if the request has a valid Pravega version
-	normRequestVersion, err := util.NormalizeVersion(requestVersion)
-	if err != nil {
-		return fmt.Errorf("request version is not in valid format: %v", err)
-	}
-	if _, ok := supportedVersions[normRequestVersion]; !ok {
-		return fmt.Errorf("unsupported Pravega cluster version %s", requestVersion)
-	}
 
-	// Check if the request is an upgrade
-	found := &pravegav1alpha1.PravegaCluster{}
-	nn := types.NamespacedName{
-		Namespace: p.Namespace,
-		Name:      p.Name,
-	}
-	err = pwh.client.Get(context.TODO(), nn, found)
-	if err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("failed to obtain PravegaCluster resource: %v", err)
+	if p.Status.IsClusterInUpgradeFailedState() {
+		if requestVersion != p.Status.GetLastVersion() {
+			return fmt.Errorf("Rollback to version %s not supported. Only rollback to previous stable version is supported.", requestVersion)
```

Comment:
It would be better if we could add the previous stable version number here too.

Reply:
Done